### PR TITLE
RobustSerial ->NeoPixel

### PR DESCRIPTION
**Pull request by SentryCoderDev**
Hello, I made a few tweaks to your current Neopixel code. I wonder if you have a chance to check it out. My purpose in the code was to transfer the colors and animations to Arduino and drive the neopixel with Arduino. Frankly, I am very confused and I ask for your help.

I used the v2 branch for the changes, it is not finished yet, I need to edit the Arduino sketch and robust serial.

### CODE REVIEW

File: modules/neopx.py [hunk 1]
```diff
@@ -1,22 +1,8 @@
 from pubsub import pub
+from modules.config import Config
+from modules.arduinoserial import ArduinoSerial
 from time import sleep
-#from colour import Color
```
Comment (danic85) [inline modules/neopx.py:3]: We can't remove this because this is the default configuration.

File: modules/neopx.py [hunk 2]
```diff
@@ -49,20 +35,6 @@ def __init__(self, count, **kwargs):
         self.animation = False
         self.thread = None
         self.overridden = False  # prevent any further changes until released (for flashlight)
-        if Config.get('neopixel', 'i2c'):
```
Comment (danic85) [inline modules/neopx.py:52]: Again, this can't be removed.

File: modules/neopx.py [hunk 3]
```diff
@@ -129,16 +104,14 @@ def set(self, identifiers, color):
...
                     pub.sendMessage('log', msg='[LED] Error in set pixels: index out of range')
                     print('Error in set pixels: index out of range')
-                    i = self.count-1                
+                    i = self.count-1
                 self.pixels[i] = color
             except Exception as e:
                 print(e)
                 pub.sendMessage('log', msg='[LED] Error in set pixels: ' + str(e))
                 pass
+        pub.sendMessage('serial', ArduinoSerial.DEVICE_LED, identifiers, color)
```
Comment (danic85) [inline modules/neopx.py:113]: This needs to be in a condition, similar to `if Config.get('neopixel', 'i2c'):` in the code above, so you can define whether to use serial coms for this in the config.json.

File: modules/neopx.py [hunk 4]
```diff
@@ -129,16 +104,14 @@ def set(self, identifiers, color):
...
         
-        self.pixels.show()
```
Comment (danic85) [inline modules/neopx.py:139]: Can't remove this.

**Comment by danic85** [comment]
I can see what you're going for, the serial command is the only change needed here to enable your functionality, and you should be able to toggle the 'mode' to be serial, gpio or i2c. 

I don't mind including this functionality again in the new neopixel module I'm making for viam, but I can't merge this until it is compatible with the other existing modes, rather than replacing them.

**Comment by SentryCoderDev** [comment]
oh I understand what you mean @danic85  , I wrote the code to work alone (directly) without I2C or GPIO (to understand how it will work). I will return it to its previous state when the warehouse draft merge request.

**Comment by SentryCoderDev** [comment]
We currently have the opportunity to choose configuration for neopixel in `config.json` but arduino it's not completed yet. 

We need to update the `case LED:` definition in the Arduino code according to v4.

**Comment by SentryCoderDev** [comment]
> I can see what you're going for, the serial command is the only change needed here to enable your functionality, and you should be able to toggle the 'mode' to be serial, gpio or i2c. 
> 
> I don't mind including this functionality again in the new neopixel module I'm making for viam, but I can't merge this until it is compatible with the other existing modes, rather than replacing them. 

Uh, do I need to set PR to Viam branch instead of V4, if I don't misunderstand?

**Comment by danic85** [comment]
> > I can see what you're going for, the serial command is the only change needed here to enable your functionality, and you should be able to toggle the 'mode' to be serial, gpio or i2c.
> > I don't mind including this functionality again in the new neopixel module I'm making for viam, but I can't merge this until it is compatible with the other existing modes, rather than replacing them.
> 
> Uh, do I need to set PR to Viam branch instead of V4, if I don't misunderstand?
No, I will add changes to the viam modules. They aren't part of the Viam branch. I will be closing that branch soon.

**Pull request closed without merging**